src/imitation/data/rollout.py

```diff
@@ -366,43 +376,57 @@ def mean_return(*args, **kwargs) -> float:
     return rollout_stats(trajectories)["return_mean"]
 
 
+T = TypeVar("T", bound=types.TransitionsMinimal)
+
+
 def flatten_trajectories(
```

Comment:
I don't get what we're trying to do here.

`TransitionsWithRew` is a `Transitions` is a `TransitionsMinimal` -- so why do we need `transitions_cls`? It's fine to return a `Transitions` and use it as a `TransitionsMinimal`. I guess it might take up some extra space?

I get the desire to not have both `flatten_trajectories` and `flatten_trajectories_with_rew`, but combining them doesn't seem to reduce code complexity overall, and it loses some typing guarantees (since we now have to do a run-time check that "rews" is present).

I may be missing something but right now I'd be in favor of reverting this change.

Reply:
I think I built this because I wanted a way to generate `TransitionsMinimal` from `List[Trajectory]`. Agreed that `Transitions` can just be substituted for `TransitionsMinimal`, so I'll probably revert this.

One way we could keep all static type checking guarantees with this merged function is using `@typing.overload`.

Reply:
Reverted.